modules/default.nix

```diff
@@ -0,0 +1,75 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  moduleName = "boulette";
+  cfg = config.services.${moduleName};
+in {
+  options.services.${moduleName} = {
+    enable = mkEnableOption "Enable ${moduleName}";
+    sshOnly = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Boulette confirmation prompt will be triggerd inside ssh session only.";
+    };
+    challengeType = mkOption {
+      type = types.string;
+      default = "ask";
+      example = "numbers";
+      description = ''
+        One of type:
+        - "ask": (default) You have to type 'y' or 'n' to resume commande execution.
+        - "hostname": You must type the host name to resume command execution.
+        - "numbers": You must type a random 6 number sequence to resume command execution.
+      '';
+    };
+    enableZsh = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable ${moduleName} for `zsh` shell";
+    };
+    enableBash = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable ${moduleName} for `bash` shell";
+    };
+  };
+
+  config = let
+    boulette = pkgs.callPackage ./../package.nix {};
+    shutdownFunction = let
+      sshOnly =
+        if cfg.sshOnly == true
+        then "--ssh-only"
+        else "";
+      challengeType =
+        if cfg.challengeType != "ask" # Remember we default to "ask"
+        then "--challenge ${cfg.challengeType}"
+        else "";
+    in ''
+      # From ${moduleName}
+      shutdown () {
+        ${boulette}/bin/boulette "shutdown $argv" ${sshOnly} ${challengeType}
+      }
+
+      # From ${moduleName}
+      reboot () {
+        ${boulette}/bin/boulette reboot ${sshOnly} ${challengeType}
+      }
+    '';
```

Comment:
We could make a nicely mapped function to generate custom checks; the api could be:

```nix
customChecks = {
  functionName = {
    command = "restart server";
    challenge = "ask";
    sshOnly = true;
  }
};
```
which would result in something like this:
```bash
functionName () {
   BOULETTEPATH/bin/boulette "restart server" --ssh-only --challenge ask
}

```

Reply:
Save this idea for later for when I have more time.